xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/FormContainerElement.java

```diff
@@ -83,10 +83,12 @@ public void fillFieldsByName(Map<String, String> valuesByNames)
         /* Register password confirmation is usually the last element that needs to be validated by liveValidation.
           This wait allows to solve a race condition between the form submission and the computation of the status of
           those fields. We force the status to be solved before we try anything else, especially submitting the form.
+          Unfortunately in Java17 we do not have lastEntry(), so we use a few non optimized operations instead. 
+          This is okay because the Map should not contain a lot of elements.
           */
-        if (valuesByNames.containsKey("register2_password")) {
-            getDriver().waitUntilCondition(driver -> !getFormElement().findElement(By.name("register2_password"))
-                .getAttribute(CLASS_ATTRIBUTE).isEmpty());
+        if(valuesByElements.size() > 0) {
+            WebElement lastElement = ((WebElement)valuesByElements.keySet().toArray()[valuesByElements.size() - 1]);
```

Comment:
Instead of doing this complex computation, why not just keeping the element in the for-loop above when iterating on the map, and reusing that element?

Reply:
I could not figure out a way to know when this is the last passage in the loop. I wanted to use a condition on .next() but it didn't go anywhere. I figured out that it was cleaner to have a one liner like this rather than have X assignations of the same variable. 

I updated this in https://github.com/xwiki/xwiki-platform/pull/3675/commits/22aebacc8cf2b8325c616b122595ffaef5c85bae :+1: